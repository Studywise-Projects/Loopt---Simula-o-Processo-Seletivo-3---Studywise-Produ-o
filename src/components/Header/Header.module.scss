@import '@/styles/_global.scss';

.toolbar {
  align-items: center;
  background-color: $primary-color;
  display: flex;
  gap: 16px;
  height: 60px;
  padding-left: 16px;

  @media only screen and (min-width: 768px) {
    height: 70px;
    padding-left: 32px;
  }
}

.iconButton {
  color: $white;
}