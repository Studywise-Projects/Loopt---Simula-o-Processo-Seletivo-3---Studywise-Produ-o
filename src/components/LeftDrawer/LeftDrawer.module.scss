@import '@/styles/_global.scss';

.swipeableDrawer {
  padding: 32px;
}

.iconButton {
  color: $white;
  padding: 0;
}

.link {
  text-decoration: none;
}

.link:active {
  text-decoration: none;
  font-weight: 700;
}

.listItem {
  padding: 8px 0;
}

.listIcon {
  color: $black;
}

.listIcon:active {
  font-weight: 700;
}

.listText {
  color: $black;
}

.listText:active {
  font-weight: 700;
}