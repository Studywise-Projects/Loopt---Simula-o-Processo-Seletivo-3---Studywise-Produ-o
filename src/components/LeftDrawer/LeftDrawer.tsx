import React from 'react';
import {
  IconButton,
  List,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  SwipeableDrawer,
} from '@mui/material';
import { useState } from 'react';
import MenuIcon from '@mui/icons-material/Menu';
import HomeIcon from '@mui/icons-material/Home';
import StarIcon from '@mui/icons-material/Star';
import PeopleIcon from '@mui/icons-material/People';
import Logo from '../Logo/Logo';
import styles from './LeftDrawer.module.scss';
import SelectAutoComplete from '../SelectAutoComplete/SelectAutoComplete';
import Link from 'next/link';

interface ILeftDrawer {
  optionsSelect: any;
  valueSelect: Object | null;
  handleChangeSelect: any;
}

function LeftDrawer({
  optionsSelect,
  valueSelect,
  handleChangeSelect,
}: ILeftDrawer) {
  const [state, setState] = useState(false);

  const toggleDrawer =
    (open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => {
      if (
        event &&
        event.type === 'keydown' &&
        ((event as React.KeyboardEvent).key === 'Tab' ||
          (event as React.KeyboardEvent).key === 'Shift')
      ) {
        return;
      }

      setState(open);
    };

  return (
    <>
      <IconButton onClick={toggleDrawer(true)} className={styles.iconButton}>
        <MenuIcon sx={{ fontSize: '32px' }} />
      </IconButton>
      <SwipeableDrawer
        anchor='left'
        open={state}
        onClose={toggleDrawer(false)}
        onOpen={toggleDrawer(true)}
        className={styles.swipeableDrawer}
        sx={{
          '& .MuiPaper-root': {
            gap: '16px',
            padding: '32px',
          },
          '& .MuiDrawer-paper': {
            boxSizing: 'border-box',
            paddingTop: '16px',
            width: '300px',
          },
        }}
      >
        <Logo variant='horizontal' />
        <SelectAutoComplete
          options={optionsSelect}
          label='Vaga'
          value={valueSelect}
          handleChange={handleChangeSelect}
          isAuxSelect={true}
        />
        <List>
          <Link href='/' className={styles.link}>
            <ListItemButton className={styles.listItem}>
              <ListItemIcon className={styles.listIcon}>
                <HomeIcon />
              </ListItemIcon>
              <ListItemText primary='Home' className={styles.listText} />
            </ListItemButton>
          </Link>

          <Link href='/' className={styles.link}>
            <ListItemButton className={styles.listItem}>
              <ListItemIcon className={styles.listIcon}>
                <StarIcon />
              </ListItemIcon>
              <ListItemText
                primary='Candidato Escolhido'
                className={styles.listText}
              />
            </ListItemButton>
          </Link>

          <Link href='/' className={styles.link}>
            <ListItemButton className={styles.listItem}>
              <ListItemIcon className={styles.listIcon}>
                <PeopleIcon />
              </ListItemIcon>
              <ListItemText
                primary='Todos os Candidatos'
                className={styles.listText}
              />
            </ListItemButton>
          </Link>
        </List>
      </SwipeableDrawer>
    </>
  );
}

export default LeftDrawer;
