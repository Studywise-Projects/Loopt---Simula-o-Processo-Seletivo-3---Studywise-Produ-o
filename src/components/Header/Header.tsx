import { AppBar, IconButton, Toolbar } from '@mui/material';
import Typography from '../Typography/Typography';
import styles from './Header.module.scss';
import LeftDrawer from '../LeftDrawer/LeftDrawer';
import useJobsStore from '../../stores/jobs';
import { ArrowBack } from '@mui/icons-material';
import { useRouter } from 'next/router';

interface IHeader {
  variant?: 'main' | 'return';
  text: string;
}

function Header({ variant = 'main', text }: IHeader) {
  const [jobs, selectedJob, setSelectedJob] = useJobsStore((state) => [
    state.jobs,
    state.selectedJob,
    state.setSelectedJob,
  ]);
  const handleChangeSelectJob = (event: any) =>
    setSelectedJob(event.target.value);

  const router = useRouter();

  return (
    <>
      {variant === 'main' ? (
        <AppBar>
          <Toolbar className={styles.toolbar}>
            <LeftDrawer
              optionsSelect={jobs}
              valueSelect={selectedJob}
              handleChangeSelect={handleChangeSelectJob}
            />
            <Typography variant='header' text={text} />
          </Toolbar>
        </AppBar>
      ) : (
        <AppBar>
          <Toolbar className={styles.toolbar}>
            <IconButton onClick={router.back} className={styles.iconButton}>
              <ArrowBack className={styles.icon} />
            </IconButton>
            <Typography variant='header' text={text} />
          </Toolbar>
        </AppBar>
      )}
    </>
  );
}

export default Header;
