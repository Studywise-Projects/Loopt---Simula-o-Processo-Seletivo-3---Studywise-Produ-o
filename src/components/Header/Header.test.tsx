import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Header from './Header';
import LeftDrawer from '../LeftDrawer/LeftDrawer';

describe('Header', () => {
  it('Should match the snapshot', () => {
    const wrapper = shallow(
      <Header
        icon={
          <LeftDrawer
            optionsSelect={undefined}
            valueSelect={null}
            handleChangeSelect={() => undefined}
          />
        }
        text='Header'
        handleClick={() => undefined}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Should call handleClickMock function once', () => {
    const handleClickMock = jest.fn();

    const wrapper = shallow(
      <Header
        icon={
          <LeftDrawer
            optionsSelect={undefined}
            valueSelect={null}
            handleChangeSelect={() => undefined}
          />
        }
        text='Header'
        handleClick={handleClickMock}
      />,
    );

    wrapper.find('[data-testId="header"]').simulate('click');

    expect(handleClickMock).toHaveBeenCalledTimes(1);
  });
});
