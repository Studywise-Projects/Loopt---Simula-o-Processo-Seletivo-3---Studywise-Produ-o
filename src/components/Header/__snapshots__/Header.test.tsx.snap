// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match the snapshot 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 undefined css-ahj2mt-MuiTypography-root"
>
  Header
</p>
`;
